dlt/common/schema/typing.py

```diff
@@ -62,6 +62,15 @@ class TColumnSchema(TColumnSchemaBase, total=False):
 TColumnName = NewType("TColumnName", str)
 SIMPLE_REGEX_PREFIX = "re:"
 
+TSchemaEvolutionMode = Literal["evolve", "freeze-and-trim", "freeze-and-raise", "freeze-and-discard"]
```

Comment:
I think those names do not explain what happens
* evolve - lets keep it for now
* freeze-and-trim -> discard-value
* freeze-and-raise -> freeze
* freeze-and-discard -> discard-row
WDYT?